tsdb/head_append.go

```diff
@@ -978,23 +978,39 @@ func exemplarsForEncoding(es []exemplarWithSeriesRef) []record.RefExemplar {
 	return ret
 }
 
-// Commit writes to the WAL and adds the data to the Head.
-// TODO(codesome): Refactor this method to reduce indentation and make it more readable.
-func (a *headAppender) Commit() (err error) {
-	if a.closed {
-		return ErrAppenderClosed
-	}
-	defer func() { a.closed = true }()
-
-	if err := a.log(); err != nil {
-		_ = a.Rollback() // Most likely the same error will happen again.
-		return fmt.Errorf("write to WAL: %w", err)
-	}
-
-	if a.head.writeNotified != nil {
-		a.head.writeNotified.Notify()
-	}
+type appenderCommitContext struct {
```

Comment:
Hmmm, I'm not certain about the naming here but I'm also struggling to come up with something better 😅

Reply:
maybe commitStatistics? does that make sense?

Reply:
We could even split this object in two or three:
1. Holds information that will be passed to metrics, e.g. `floatsAppended`, `histogramsAppended`, etc
2. Things related to OOO records, like oooMin/MaxT, wblSample/Histograms, records, mmapMarkers, etc

Reply:
Yes, I feel you, this name is not the best one, I was struggling to get something better as well.
I'm not sure also about `commitStatistics` because seems to me this object is holding much more information than the ones will be used to increment the metrics.

About split into sub structs, I don't see much value beyond making life easier to naming the root object, did you see anything else beyond that?

Reply:
Maybe `commitMetrics`? 

NAMING! (insert old man yelling gif)